charmhelpers/contrib/openstack/ha/utils.py

```diff
@@ -93,50 +185,75 @@ def update_dns_ha_resource_params(resources, resource_params,
             status_set('blocked', msg)
             raise DNSHAException(msg)
 
-        hostname_key = 'res_{}_{}_hostname'.format(charm_name(), endpoint_type)
+        hostname_key = 'res_{}_{}_hostname'.format(service, endpoint_type)
         if hostname_key in hostname_group:
             log('DNS HA: Resource {}: {} already exists in '
                 'hostname group - skipping'.format(hostname_key, hostname),
                 DEBUG)
             continue
 
         hostname_group.append(hostname_key)
-        resources[hostname_key] = crm_ocf
-        resource_params[hostname_key] = (
-            'params fqdn="{}" ip_address="{}" '
+        relation_data['resources'][hostname_key] = crm_ocf
+        relation_data['resource_params'][hostname_key] = (
+            'params fqdn="{}" ip_address="{}"'
             ''.format(hostname, resolve_address(endpoint_type=endpoint_type,
```

Comment:
Shouldn't need the `''` on this line.  i.e. just the `.format(hostname, ...)

(I know it's not part of the change, but an opportunity to clean up?)

Reply:
Thanks for the review; update incoming.